database/tables/instance_questions.pg

```diff
@@ -13,6 +13,8 @@ columns
     incremental_submission_points_array: double precision[]
     incremental_submission_score_array: double precision[]
     last_submission_score: double precision
+    manual_grading_conflict: boolean default false
+    manual_grading_user: bigint
```

Comment:
I think this is a good idea. We can do this superficially on the front-end (do not display a warning if older than 1 hour). Or we can incorporate this logic into the next random question selection to seek out and clear out items older than 1 hour.

I avoided this logic, as I am not too sure about 1 hour being long enough. ie. lunch breaks, very advanced questions, etc. It still seems intuitively reasonable though.

Reply:
I wonder if we could put this date on the same table tentatively mentioned above? 

ie. `users_maual_grading`
```user_id, instance_question, date_started```

Reply:
I might just go ahead and try this on a new branch. The query logic will likely be much simpler.

Reply:
So the query logic is much simpler but the table misses the ability to store the manual_grading_user after the grading is done. As each user is allowed one instance question, the table is insufficient to store history. 

We may want to re-visit the possibility of a new table while integrating Maja's changes. Once I start on that, I will have a clearer understanding of how we can simplify the storage of manual grading data. It may still make the most sense to store some of the data on the instance_question for the time being.

Reply:
https://github.com/PrairieLearn/PrairieLearn/compare/manual-grading-anti-ta-overlap-spin-lock-views...manual-grading-user-manual-grading-table?expand=1

Implemented this part way. We can do this change at a later time once we have an understanding of Maja's logic. Perhaps we can re-model this further.

Reply:
The table ended up looking as such: 

```
CREATE TABLE IF NOT EXISTS users_manual_grading (
  user_id BIGINT REFERENCES users ON DELETE SET NULL ON UPDATE CASCADE,
  instance_question_id BIGINT REFERENCES instance_questions ON DELETE SET NULL ON UPDATE CASCADE,
  date_started TIMESTAMP WITH TIME ZONE,
  last_date_accessed TIMESTAMP WITH TIME ZONE DEFAULT NULL,
  date_graded TIMESTAMP WITH TIME ZONE DEFAULT NULL,
  PRIMARY KEY(user_id, instance_question_id)
);

We do record the last_date_accessed, but we do not use it for any real feature at the moment. The composite key is a way to ensure we do not enter a row each time a user accesses a question for manual grading. We upsert and update the last_accessed field if the user has already accessed this area. I am assuming we do not need to record a row for each time a user accesses an instance question for manual grading.